site/docs/v1/tech/oauth/url-validation.adoc

```diff
@@ -0,0 +1,274 @@
+
+|\https://example.com?*=blah
+|✕
+|Wildcards are not allowed in query string names
+
+|===
+
+== Wildcard Replacement Patterns
+
+The position where wildcards are allowed in configured values is just one half of the puzzle. Wildcards in each portion of the URL have different rules for the replacement values. Please note that allowed replacement values may not produce a valid URL. This section provides details on the allowed replacements for wildcards in each portion of the URL. A wildcard in the configured value must be replaced with at least one character in the concrete value.
+
+=== Domain
+
+Replacements for wildcards in the domain portion of the URL must not contain `.`, `:`, `/`, or `?` characters. The following table lists examples of valid and invalid replacements for valid wildcard patterns.
+
+[cols="25m,25m,10,40"]
+.Domain wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://*.example.com
+|\https://login.example.com
+|✓
+|&nbsp;
+
+|\https://auth*.example.com
+|\https://auth2.example.com
+|✓
+|&nbsp;
+
+|\https://auth*.example.com
+|\https://auth.example.com
+|✕
+|The value does not contain a character to replace the `*`
+
+|\https://*.example.com
+|\https://auth.customer.example.com
+|✕
+|The replacement contains a `.` character
+
+|===
+
+=== Port
+
+Replacement values for wildcards in the port portion of the URL must consist of one or more digits.
+
+[cols="25m,25m,10,40"]
+.Port wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://example.com:*
+|\https://example.com:2012
+|✓
+|&nbsp;
+
+|\https://example.com:*
+|\https://example.com:80b
+|✕
+|The replacement value contains a non-numeric character
+
+|===
+
+=== Path Segments
+
+Replacement values for wildcards in a path segment of the URL must not contain `/` or `?` characters.
+
+[cols="25m,25m,10,40"]
+.Path segment wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://example.com/path/*/resource
+|\https://example.com/path/to/resource
+|✓
+|&nbsp;
+
+|\https://example.com/p*/to/resource
+|\https://example.com/path/to/resource
+|✓
+|&nbsp;
+
+|\https://example.com/pass:[*]/par*tial/pass:[*]
+|\https://example.com/path/partotial/resource
+|✓
+|&nbsp;
+
+|\https://example.com/path/*/resource
+|\https://example.com/path/to/the/resource
+|✕
+|The replacement value contains a `/`
+
+|\https://example.com/path/*
+|\https://example.com/path/resource?foo=bar
+|✕
+|The replacement value contains a `?`
+
+|\https://example.com/pass:[*]/par*tial/pass:[*]
+|\https://example.com/path/partial/resource
+|✕
+|The segment `partial` does not contain a replacement character for the wildcard
+
+|===
+
+=== Query String Values
+
+Replacement values for query string values must not contain the `&` character.
+
+[cols="25m,25m,10,40"]
+.Query string value wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://example.com?foo=*
+|\https://example.com?foo=bar
```

Comment:
Does this match `https://example.com?bar=baz&foo=bar`?

I'd guess no, since `bar=baz` is not matched.

Reply:
That's correct. Added an example.